web/book/tests/snapshots/snapshot__tests__prql__language-features__f-strings-1.prql.snap

```diff
@@ -1,16 +1,19 @@
 ---
-source: book/tests/snapshot.rs
+source: web/book/tests/snapshot.rs
 expression: "from web\nselect url = f\"http{tls}://www.{domain}.{tld}/{page}\"\n"
 ---
 SELECT
   CONCAT(
-    'http',
-    tls,
-    '://www.',
-    domain,
-    '.',
-    tld,
-    '/',
+    CONCAT(
+      CONCAT(
+        CONCAT(
+          CONCAT(CONCAT(CONCAT('http', tls), '://www.'), domain),
+          '.'
+        ),
+        tld
+      ),
+      '/'
+    ),
```

Comment:
How viable would it be to have `CONCAT` be a variarg function? This isn't the friendliest SQL. (But not a showstopper by any means...)

I guess it's quite difficult at the moment given the proposed code has the number of args in the compiler (https://github.com/PRQL/prql/pull/2428/files#diff-2540aa8b42227cb2ab1983dcc56db5f2401778a2ba89fb1c33525c31c59d7e6bR97) , and we don't yet support lists?

Reply:
I've intentionally made functions have predefined number and types of arguments. But limitation is imposed on RQ only, we can do some clever translations in sql backend to output a single CONCAT function call.